index.js

```diff
@@ -6,30 +6,16 @@ var xtend = require("xtend")
 
 module.exports = createXHR
 createXHR.XMLHttpRequest = window.XMLHttpRequest || noop
-createXHR.XDomainRequest = "withCredentials" in (new createXHR.XMLHttpRequest()) ? createXHR.XMLHttpRequest : window.XDomainRequest
 createXHR.qsSerialize = null // Define this as a function to support the `qs` option
 
-forEachArray(["get", "put", "post", "patch", "head", "delete"], function(method) {
+"get,put,post,patch,head,delete".split(",").map(function(method) {
```

Comment:
@jmeas an ES6 implementation would be generally a lot smaller, but then the only reason to use this instead of `.fetch` would be to get an xhr that you can abort. Not sure if that's enough value proposition to get any adoption.
Support for not-so-old browsers is the key here.

Anyway, xhr.modern or xhr.es6 is a good idea to introduce next, once we get v3 shipped.